webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/hooks/RunHooks.scala

```diff
@@ -304,42 +323,77 @@ object RunHooks {
    * Hooks must be executable and not ends with one of the
    * non-executable extensions.
    */
-   def getHooks(basePath: String, ignoreSuffixes: List[String]): Box[Hooks] = getHooksPure(basePath, ignoreSuffixes).toBox
-
-   def getHooksPure(basePath: String, ignoreSuffixes: List[String]): IOResult[Hooks] = {
-     IOResult.effect {
-       val dir = new File(basePath)
-       // Check that dir exists before looking in it
-       if (dir.exists) {
-         HooksLogger.debug(s"Looking for hooks in directory '${basePath}', ignoring files with suffix: '${ignoreSuffixes.mkString("','")}'")
-         // only keep executable files
-         val files = dir.listFiles().toList.flatMap { file =>
-           file match {
-             case f if (f.isDirectory) => None
-             case f =>
-               if(f.canExecute) {
-                 val name = f.getName
-                 //compare ignore case (that's why it's a regienMatches) extension and name
-                 ignoreSuffixes.find(suffix => name.regionMatches(true, name.length - suffix.length, suffix, 0, suffix.length)) match {
-                   case Some(suffix) =>
-                     HooksLogger.debug(s"Ignoring hook '${f.getAbsolutePath}' because suffix '${suffix}' is in the ignore list")
-                     None
-                   case None      =>
-                     Some(f.getName)
-                 }
-               } else {
-                 HooksLogger.debug(s"Ignoring hook '${f.getAbsolutePath}' because it is not executable. Check permission if not expected behavior.")
-                 None
-               }
-           }
-         }.sorted // sort them alphanumericaly
-         Hooks(basePath, files)
-       } else {
-         HooksLogger.debug(s"Ignoring hook directory '${dir.getAbsolutePath}' because path does not exists")
-         // return an empty Hook
-         Hooks(basePath, List[String]())
-       }
-     }
-   }
+  def getHooks(basePath: String, ignoreSuffixes: List[String]): Box[Hooks] = getHooksPure(basePath, ignoreSuffixes).toBox
+
+  /*
+   * Check for the existence of hook limit in the 10 first lines of the file
```

Comment:
i don't see where the 10 lines limits comes from

Reply:
right, I changed it to 1000 chars (more binary-file resilient)